File: 0045-jump-game-ii/0045-jump-game-ii.cpp
class Solution {
public:
    int jump(vector<int>& nums) {
        int L = 0;
        int R = 0;
        int result = 0;

        while(R < size(nums)-1){
            int farthest = 0;
            for(int i = L; i <= R; i++){
                farthest = max(farthest, i + nums[i]);
            }
            L = R + 1;
            R = farthest;
            result++;
        }

        return result;
    }
};

File: 0045-jump-game-ii/0045-jump-game-ii.py
class Solution(object):
    def jump(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        L, R = 0, 0
        farthest = 0
        count = 0

        while R < len(nums)-1:
            for i in range(L, R+1):
                farthest = max(farthest, nums[i] + i)
            
            L = R+1
            R = farthest
            count += 1
        
        return count
